neqo-transport/src/path.rs

```diff
@@ -331,8 +330,10 @@ impl Paths {
                     .as_ref()
                     .is_some_and(|target| Rc::ptr_eq(target, p))
                 {
-                    let primary = self.migration_target.take();
-                    mem::drop(self.select_primary(&primary.unwrap(), now));
+                    let Some(primary) = self.migration_target.take() else {
+                        break;
```

Comment:
I think that you can restructure this little piece of code to avoid the extra break.

```rust
let Some(primary) = self.migration_target.as_mut() else { break; }
if Rc::ptr_eq(target, primary) {
  let p = primary.take();
  mem::drop(self.select_primary(&p, now));
  return true;
}
break;
```

Once we're on 1.80 MSRV, we can use `Option::take_if` and simplify further.

```rust
if let Some(primary) = self.migration_target.take_if(|p| Rc::ptr_eq(target, p)) 
  mem::drop(self.select_primary(&primary, now));
  return true;
}
break;
```